src/workers/event.rs

```diff
@@ -556,3 +619,892 @@ impl<S: Store + Send + Sync> Worker for EventWorker<S> {
         message.ack().await.map_err(WorkError::from)
     }
 }
+
+#[cfg(test)]
+mod test {
+    use std::sync::Arc;
+
+    use super::*;
+
+    use crate::test_util::TestStore;
+
+    #[async_trait::async_trait]
+    impl ClaimsSource for HashMap<String, Claims> {
+        async fn get_claims(&self) -> anyhow::Result<HashMap<String, Claims>> {
+            Ok(self.clone())
+        }
+    }
+
+    // NOTE: This test is rather long because we want to run through what an actual state generation
+    // loop would look like. This mostly covers happy path, while the other tests cover more of the
+    // edge cases
+    #[tokio::test]
+    async fn test_all_state() {
+        let store = Arc::new(TestStore::default());
+        let worker = EventWorker::new(store.clone(), HashMap::default());
+
+        let lattice_id = "all_state";
+        let host1_id = "DS1".to_string();
+        let host2_id = "starkiller".to_string();
+
+        /***********************************************************/
+        /******************** Host Start Tests *********************/
+        /***********************************************************/
+        let labels = HashMap::from([("superweapon".to_string(), "true".to_string())]);
+        worker
+            .handle_host_started(
+                lattice_id,
+                &HostStarted {
```

Comment:
I think if we write more tests, it would be helpful to have simple functions for generating this host started event. E.g.
```
fn host_started_evt("spongebobid", "sPoNgEbObId", [("sponge".to_string(), "bob".to_string())])
```

May not end up saving too much but it's nice